olive/model/handler/snpe.py

```diff
@@ -58,7 +58,8 @@ def prepare_session(
 
     def to_json(self, check_object: bool = False):
         config = super().to_json(check_object)
-        config["config"].update(self.io_config)
+        # only update the key in io_config where the value is not None
+        config["config"].update({key: value for key, value in config["config"].items() if value})
```

Comment:
bug here.

Reply:
I think the similar issue with extra keys will happen here too https://github.com/microsoft/Olive/blob/d068cf963e0ec1c5d4ecf9b87c2a1e118fe2303f/olive/passes/snpe/quantization.py#L109. 

Maybe the `get_io_config` method for this handler can be implemented to only return the 4 valid key, values? 

and then we can just call `get_io_config()` for this handler instead of accessing `io_config` directly?

Reply:
IMHO, the line to exclude the key here is needless. or at least, it is not the right place to exclude keys. A better way is in the snpe.__init__ to remove the unexpected keys instead of base on value.

BTW, the io_config is missing after this line change.

Reply:
Then seems original logics for snpe is clear. Maybe we can just ignore this warning for snpe.
![image](https://github.com/microsoft/Olive/assets/13343117/ce5380dc-457c-4a43-9f84-52a7f4ec8c77)